### ISSUE TITLE
Transfering a pokemon messes up Pokebank

### ISSUE THREAD

**Issue by bbb0x** [Grover-c13/PokeGOAPI-Java#749]
Each time when I transfer a POKéMON with pokemon.TransferPokemon(),  Pokebank.getPokemons() returns an empty list; only POKéMONs captured after the transfer will appear in it. Restarting or checking the client shows that I did NOT accidentely nuke my team. This is a critical mistake as it messes up the entire program. Even when updating the inventories before getting the pokemon list does not help.

My call order:

```
 pkmn.transferPokemon();
 bot.GetInventories().updateInventories();
 bot.GetInventories().getPokebank().getPokemons()
```

EDIT: Looking at the source; transferPokemon calls Pokebanks RemovePokemon:

```
public void removePokemon(final Pokemon pokemon) {
        pokemons.clear();
        pokemons.addAll(Stream.of(pokemons).filter(new Predicate<Pokemon>() {
            @Override
            public boolean test(Pokemon pokemn) {
                return pokemn.getId() != pokemon.getId();
            }
        }).collect(Collectors.<Pokemon>toList()));
    }
```

Since pokemons.clear is the first line, doesn't that mean that an empty list is added on it?
Doesn't the list needs to get copied into another one, then clear the current pokemon list and add the copied one with the filters?

**Comment by bbb0x** [comment]
Fixed, I just don't know enough of github to update it:

```
public void removePokemon(final Pokemon pokemon) {
        List<Pokemon> backup = new ArrayList<Pokemon>();
        backup.addAll(pokemons);

        pokemons.clear();
        pokemons.addAll(Stream.of(backup).filter(new Predicate<Pokemon>() {
            @Override
            public boolean test(Pokemon pokemn) {
                return pokemn.getId() != pokemon.getId();
            }
        }).collect(Collectors.<Pokemon>toList()));
    }

```

**Comment by Gegy** [comment]
This also occurs when evolving a Pokemon, but that's because evolve also calls removePokemon

**Comment by Gegy** [comment]
This was fixed in #791
